app/controllers/api/cases_controller.rb

```diff
@@ -0,0 +1,51 @@
+class API::CasesController < ApplicationController
+  respond_to :json
+
+  def create
+    _case = Case.new(case_params)
+    begin
+      saved = _case.save!
+    rescue ActiveRecord::StatementInvalid => invalid
+      return render json: {message: 'Invalid case'}
+    end
+    if saved
+      return render json: {message: 'Case successfully created!',
+                           case: _case}
+    else
+      return render json: {error: _case.errors.full_messages,
+                           status: 500}
+    end
+    return render json: {message: params[:type_of_case]}
+  end
+
+  def case_params
+    params.permit(
```

Comment:
this should be `params.require(:case).permit( /* All the attributes */)`, see my comment above about the payload